### DISCUSSION TITLE
update instructions?

### DISCUSSION THREAD

**Discussion by Charmandrigo** [AbdBarho/stable-diffusion-webui-docker#544] [Q&A]
whenever there's a new release is it auto updated or it manually has to be removed and reinstalled? 
I am asking because I don't want to lose my current settings, models and extensions I currently have on my automatic1111 install

**Comment by kienerj** [comment]
Yeah having update instructions would be useful not to mess things up or lose all settings

**Reply by ryanbrackney** [reply]
It doesn't auto update, but if you pull the newest version of the repo and rebuild the container you will get an updated version (if the docker container has been advanced since you pulled last). Normally this won't override your settings. 

If you're worried about it you can always just duplicate your data directory and give it a go.

You can also revert to old versions by modifying which docker image version you're building.

**Reply by AbdBarho** [reply]
what @ryanbrackney is correct.

if you want to update just do a `git pull` and you have the latest changes, when you run again the container will be rebuilt, I will add more info to the Wiki page

**Comment by lalamax3d** [comment]
more questions about update...
i.e i git clone repo and install today. its automatic 1.5.1. i know automatic 1.6.0-RC is released and worked fine. can i just update my automatic repo somehow to that specific version / tag ?

**Reply by DevilaN** [reply]
It is RC for a reason. Release Candidate means it might be not working fine for other people than you.
You can edit `./services/AUTOMATIC1111/Dockerfile`. In line 47 there is commit hash from original A1111 repository that is being used to create docker image here, so you can use different commit hash pointing to the version you've mentioned. Rebuild image and run it, but from now one you are on your own trying to debug / find why something is not working properly despite first "worked fine" impression.
